core/src/main/java/brooklyn/entity/rebind/PeriodicDeltaChangeListener.java

```diff
@@ -78,6 +78,8 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(PeriodicDeltaChangeListener.class);
 
+    protected final AtomicInteger checkpointLogCount = new AtomicInteger();
```

Comment:
I think you should use an `AtomicLong` instead. If `shouldLogCheckpoint` were called once per millisecond the integer would overflow in about three and a half weeks: `(2^31 - 1) / (1000 * 60 * 60 * 24 * 7) = 3.55`. So `shouldLogCheckpoint` will always return true until `checkpointLogCount.get() < 5` is false. With a Long it would take  292,277 millenia to overflow.